core/src/main/java/org/apache/calcite/util/RangeSets.java

```diff
@@ -128,6 +129,14 @@ private RangeSets() {}
         && !range.isEmpty();
   }
 
+  /** Returns whether a range set is a single open interval. */
+  public static <C extends Comparable<C>> boolean isOpenInterval(RangeSet<C> rangeSet) {
+    final Set<Range<C>> ranges = rangeSet.asRanges();
+    final Range<C> range = ranges.iterator().next();
+    return ranges.size() == 1
```

Comment:
Will throw if `ranges.isEmpty()` because you iterate before you check `size`. Add a test where this is called on an empty range set.

Reply:
Agree, have added the tests.